audio_capture/src/audio_capture.cpp

```diff
@@ -51,12 +59,28 @@ namespace audio_transport
         _source = gst_element_factory_make("alsasrc", "source");
         _convert = gst_element_factory_make("audioconvert", "convert");
 
-        _encode = gst_element_factory_make("lame", "encoder");
-        g_object_set( G_OBJECT(_encode), "preset", 1001, NULL);
-        g_object_set( G_OBJECT(_encode), "bitrate", _bitrate, NULL);
-
-        gst_bin_add_many( GST_BIN(_pipeline), _source, _convert, _encode, _sink, NULL);
-        gst_element_link_many(_source, _convert, _encode, _sink, NULL);
+        if (_format == "mp3"){
+          _encode = gst_element_factory_make("lame", "encoder");
+          g_object_set( G_OBJECT(_encode), "preset", 1001, NULL);
+          g_object_set( G_OBJECT(_encode), "bitrate", _bitrate, NULL);
+
+          gst_bin_add_many( GST_BIN(_pipeline), _source, _convert, _encode, _sink, NULL);
+          gst_element_link_many(_source, _convert, _encode, _sink, NULL);
+        } else if (_format == "wave") {
+          GstCaps *caps;
+          caps = gst_caps_new_simple("audio/x-raw-int",
+                                     "channels", G_TYPE_INT, _channels,
+                                     "width",    G_TYPE_INT, _depth,
+                                     "depth",    G_TYPE_INT, _depth,
+                                     "rate",     G_TYPE_INT, _sample_rate,
+                                     "signed",   G_TYPE_BOOLEAN, TRUE,
+                                     NULL);
+
+          g_object_set( G_OBJECT(_sink), "caps", caps, NULL);
+          gst_caps_unref(caps);
+          gst_bin_add_many( GST_BIN(_pipeline), _source, _sink, NULL);
+          gst_element_link_many( _source, _sink, NULL);
+        }
```

Comment:
This needs an `else` to handle the case where the user has not set the format correctly.